toolkit/primitives/domain/src/lib.rs

```diff
@@ -258,6 +257,20 @@ const MAINCHAIN_PRIVATE_KEY_LEN: usize = 32;
 #[byte_string(hex_serialize, hex_deserialize)]
 pub struct MainchainPrivateKey(pub [u8; MAINCHAIN_PRIVATE_KEY_LEN]);
 
+impl MainchainPrivateKey {
+	#[cfg(feature = "std")]
+	pub fn sign(&self, unsigned_message: &[u8]) -> MainchainSignature {
```

Comment:
Is it used? Should not be, because this type is 32 bytes, but extended keys that are used in CLI can be 64 bytes and all signing is done there. I should be OK, to remove this addition.

MainchainPrivateKey seems to be used only in tests. As said above - it cannot be valid type because we use also 64 bytes keys.

CardanoPaymentSigningKey is used for signing in CLI.